### DISCUSSION TITLE
Ignore inserts/updates/delete from within the function that's using the SQL trigger

### DISCUSSION THREAD

**Discussion by punocchio1** [Azure/azure-functions-sql-extension#840] [Q&A]
Hi

I've written an Azure function using the SQL trigger to process records that have been inserted into a table. When I am finished with the processing, I need to update the record that was processed to indicate it was successfully processed. Currently I am using a DbContext and calling ExecuteSqlInterpolated to update the underlying record from within the function, using the ID of the record that was processed via the triggered function. However, this results in another change to the DB that causes the trigger to fire again. When the function runs, I can filter the `changes` list to only include inserts, like this: `changes.Any(m => m.Operation == SqlChangeOperation.Insert)` That way I skip processing for those, but it is an unnecessary call to the function nonetheless. Is there a way to ignore updates (or inserts or delete for that matter) that are made inside the function, either by not tracking them, or using `CHANGE_TRACKING_CONTEXT` in the `UPDATE` statement and ignoring that context in the queries in this library's class `SqlTableChangeMonitor` ? This would be similar to what's described here https://www.mssqltips.com/sqlservertip/6521/sync-sql-server-change-tracking-tables-without-changing-data/

Thanks!

**Comment by Charles-Gagnon** [comment]
Unfortunately, that isn't something that we support at this time. This seems like a very useful thing to have though, so I opened an issue to track it - https://github.com/Azure/azure-functions-sql-extension/issues/852

You can follow that for any updates, and if you have anything else to add please do so! Thank you!
